### PR TITLE
BCE - 15953 fix filtering issues

### PR DESCRIPTION
**Pull request by mayshavit**
Enable severity only when there are results in the severity
Move all filtering code of the results to a dedicated class

### CODE REVIEW

File: src/main/kotlin/com/bridgecrew/ui/CheckovToolWindowFactory.kt [hunk 1]
```diff
@@ -21,22 +22,23 @@ const val VULNERABILITIES_TAB_NAME = "Vulnerabilities"
 const val LICENSES_TAB_NAME = "Licenses"
 const val SECRETS_TAB_NAME = "Secrets"
 
+val tabNameToCategory: Map<String, Category?> = mapOf(
```
Comment (saarschweid) [inline src/main/kotlin/com/bridgecrew/ui/CheckovToolWindowFactory.kt:25]: can remain private no?

File: src/main/kotlin/com/bridgecrew/ui/topPanel/CheckovTopPanel.kt [hunk 2]
```diff
@@ -83,11 +85,11 @@ class CheckovTopPanel(val project: Project) : SimpleToolWindowPanel(true, true),
...
-        actionToolbarPanel.add(SeverityFilterButton(project, "I"));
-        actionToolbarPanel.add(SeverityFilterButton(project,"L"));
-        actionToolbarPanel.add(SeverityFilterButton(project,"M"));
-        actionToolbarPanel.add(SeverityFilterButton(project,"H"));
-        actionToolbarPanel.add(SeverityFilterButton(project,"C"));
+        actionToolbarPanel.add(SeverityFilterButton(project, "I", Severity.INFO));
+        actionToolbarPanel.add(SeverityFilterButton(project,"L", Severity.LOW));
+        actionToolbarPanel.add(SeverityFilterButton(project,"M", Severity.MEDIUM));
+        actionToolbarPanel.add(SeverityFilterButton(project,"H", Severity.HIGH));
+        actionToolbarPanel.add(SeverityFilterButton(project,"C", Severity.CRITICAL));
```
Comment (saarschweid) [inline src/main/kotlin/com/bridgecrew/ui/topPanel/CheckovTopPanel.kt:92]: we're ignoring "unknown"?

Comment (saarschweid) [reply src/main/kotlin/com/bridgecrew/ui/topPanel/CheckovTopPanel.kt:92]: also need to change `SeverityFilterActions` I think

**Review by saarschweid — APPROVED** [review]

File: src/main/kotlin/com/bridgecrew/ui/topPanel/CheckovTopPanel.kt (hunk 2 above)
Comment (mayshavit) [reply src/main/kotlin/com/bridgecrew/ui/topPanel/CheckovTopPanel.kt:92]: Changed also, thanks!

**Review by niradler — APPROVED** [review]

**Pull request merged**